pkg/controller/serverside/reconcile.go

```diff
@@ -39,3 +39,16 @@ func ReconcileObject(ctx context.Context, c client.Client, obj client.Object) er
 
 	return nil
 }
+
+// ReconcileReplaceObject completely replaces the current object with the new object during reconciliation.
+// This is intended for special use cases only as the preferred method to reconcile objects is server-side apply.
+func ReconcileReplaceObject(ctx context.Context, c client.Client, currentObj, newObj client.Object) error {
+	// MergeFrom creates a Patch that patches using the merge-patch strategy with the given object as base.
+	// When using MergeFrom, existing lists will be completely replaced by new lists.
+	err := c.Patch(ctx, newObj, client.MergeFrom(currentObj))
```

Comment:
this is not replacing right? it's doing a merge
I think we just need `client.Update`
and in that case, we probably don't need a helper method